src/coreclr/jit/assertionprop.cpp

```diff
@@ -2111,11 +2114,6 @@ void Compiler::optAssertionGen(GenTree* tree)
 {
     tree->ClearAssertion();
 
-    if (tree->gtFlags & GTF_COLON_COND)
```

Comment:
Is GTF_COLON_COND dead code that is being removed with this change?
Maybe change it to an assert here.

I don't believe that we can create an assertion that is predicated on a conditional

Reply:
We remove QMARKS in morph, so perhaps this is too aggressive. I'll restore the bail out but only do it for local assertion prop.